lighthouse-docker/README.md

```diff
@@ -0,0 +1,140 @@
+- By starting a web server, allowing you run Lighthouse using a REST API.
+
+### Using the CLI
+
+The container can be from the the CLI just like using the Lighthouse npm module. See
+Lighthouse docs for [CLI options](../#cli-options).
+
+```bash
+# Audit example.com. Lighthouse results are printed to stdout.
+docker run -it --rm --cap-add=SYS_ADMIN lighthouse_docker https://example.com
+
+# Audits example.com and saves HTML report to a file.
+docker run -it --rm --cap-add=SYS_ADMIN lighthouse_docker --quiet https://example.com > report.html
+
+# Audits example.com and saves JSON results to a file.
+docker run -it --rm --cap-add=SYS_ADMIN lighthouse_docker --quiet --output=json https://example.com > report.json
+
+# Print the Lighthouse version used by the container.
+docker run -it --rm --cap-add=SYS_ADMIN lighthouse_docker --version
+
+# Print Lighthouse help
+docker run -it --rm --cap-add=SYS_ADMIN lighthouse_docker --help
+```
+
+### Using the REST API web service
+
+The container also bundles a web server that supports a REST API. You can
+use it to run Lighthouse return scores...in the cloud!
+
+To run the web server, invoke `docker run` without any arguments (e.g. no `CMD`):
+
+```bash
+docker run -dit -p 8080:8080 --rm --name lighthouse_docker --cap-add=SYS_ADMIN lighthouse_docker
+```
+
+There are also npm script helpers for starting and building + restarting the server:
+
+```bash
+yarn serve
+# or
+# npm run serve
+
+# Re-build and start the server again.
+yarn restart
+# or
+# npm run restart
+```
+
+This starts a server on `8080` and exposes a REST endpoint at `http://localhost:8080/audit`.
+
+**Examples**
+
+The endpoint supports `GET` requests with the following URL parameters:
+
+- `url`: parameter for the page you want to test.
+- `output`: report format. One of `json` or `html`.
+- `nolog`: By default, logs are streamed as Lighthouse audits the page.
+   Once the report is ready, the page redirects to the final report. If you
+   don't want this feature or your hosting env doesn't support [Server
+   Sent Events](https://www.html5rocks.com/en/tutorials/eventsource/basics/),
+   include the `nolog` parameter.
+
+```bash
+http://localhost:8080/audit?url=https://example.com&output=html
+http://localhost:8080/audit?url=https://example.com&output=json
+http://localhost:8080/audit?url=https://example.com&output=json&nolog
+```
+
+The endpoint also supports `POST` requests with the same parameters. The only
+difference is that `POST` requests do not stream logs.
+
+Instead of URL query parameters, send JSON with the same param names:
+
+```bash
+curl -X POST \
+  -H "Content-Type: application/json" \
+  --data '{"output": "json", "url": "https://example.com"}' \
+  http://localhost:8080/audit
+```
+
+## Using full Chrome instead of headless Chrome
+
+By default, the `Dockerfile` launches headless Chrome to run Lighthouse. If you
+want to use full "headful" Chrome, build the image using `Dockerfile.headful`.
```

Comment:
QQ: why would you want to do this?

Reply:
I asked the same question to @brendankenny yesterday. At this point, it's mainly around for historical reasons. Headless Chrome didn't have network emulation so this was the setup for many moons.

Possible reasons to keep:
- Setting up xvfb with lighthouse is painful enough that an example is worth 1000 words. Although this could just be a section in the readme.
- GPU isn't available in headless chrome. I'm not sure how docker plays with the host GPU, but folks may want an actual display for testing.
- Other differences between headless and headful that we're not aware of.

...though I'm not convinced any of these justify us maintaining something separate.

Thoughts?

Reply:
Gotcha. Don't have any opinions here, just trying to understand what our guidance is or should be. It might be worth documenting why this exists and where and why you'd use it.

Reply:
it would be good to be able to quantify the difference. I was saying to @ebidel that if I were not involved in any of this my instinct for how to get the most "realistic" numbers would be to run headful, mostly for a vague notion of missing anything related to rendering/painting.

But I don't have even ballpark figures to back that up :)

Reply:
@brendankenny @ebidel should I do a test run on like 30 sites to measure differences?

Reply:
@brendankenny I wonder how much of that is because you're on the inside  and know about the historical issues.

Another way to look at this is: users won't know how Chrome is being run when they write `FROM lighthouse-docker` in their `Dockerfile`. Nor will/should they care if things JustWork.